pkg/server/singleprocess/service_pipeline.go

```diff
@@ -26,6 +26,45 @@ func (s *Service) UpsertPipeline(
 	return &pb.UpsertPipelineResponse{Pipeline: result}, nil
 }
 
+// GetPipeline returns a Pipeline based on a pipeline ref id
+func (s *Service) GetPipeline(
+	ctx context.Context,
+	req *pb.GetPipelineRequest,
+) (*pb.GetPipelineResponse, error) {
+	if err := serverptypes.ValidateGetPipelineRequest(req); err != nil {
+		return nil, err
+	}
+
+	p, err := s.state(ctx).PipelineGet(req.Pipeline)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.GetPipelineResponse{
+		Pipeline: p,
+		//RootStep: "", // TODO figure out root step
```

Comment:
Might as well fill this in here, rebase on f-pipelines and use `ptypes.PipelineGraph`, then call `KahnSort`, and its element 0.